### PR TITLE
Check logs in external plugins integration tests

### PR DESCRIPTION
**Pull request by sfc-gh-astus**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [ ] I've described my changes in the section below.

### Changes description
...

### CODE REVIEW

File: tests_integration/testing_utils/assertions/test_result_assertions.py [hunk 1]
```diff
@@ -5,8 +5,10 @@
 from tests_integration.test_utils import contains_row_with
 
 
-def assert_that_result_is_successful(result: CommandResult) -> None:
+def assert_that_result_is_successful(result: CommandResult, caplog=None) -> None:
```
Comment (sfc-gh-astus) [inline tests_integration/testing_utils/assertions/test_result_assertions.py:8]: Should I add that way? WDYT?

File: tests_integration/testing_utils/assertions/test_result_assertions.py [hunk 2]
```diff
@@ -5,8 +5,10 @@
...
     assert result.exit_code == 0, result.output
+    if caplog:
+        _assert_that_no_error_logs(caplog)
```
Comment (sfc-gh-pjob) [inline tests_integration/testing_utils/assertions/test_result_assertions.py:11]: IMO it should not be a part of `assert_that_result_is_successful`. Presence of error logs does not mean that result is automatically a failure.

For example running `snow sql -q "select 1"` using CLI with broken external plugin will still finish successfully.

IMO `assert_that_no_error_logs` should be a separate assertion.

File: tests_integration/testing_utils/assertions/test_result_assertions.py [hunk 3]
```diff
@@ -40,3 +42,11 @@ def assert_that_result_is_successful_and_done_is_on_output(
...
     assert result.output is not None
     assert json.loads(result.output) == {"message": "Done"}
+
+
+def _assert_that_no_error_logs(caplog):
+    error_logs = [
+        record.message for record in caplog.records if record.levelname == "ERROR"
+    ]
+    assert error_logs == []
+    caplog.clear()
```
Comment (sfc-gh-pjob) [inline tests_integration/testing_utils/assertions/test_result_assertions.py:52]: IMO we should change test's state as part of the assertion.

Comment (sfc-gh-astus) [reply tests_integration/testing_utils/assertions/test_result_assertions.py]: Removed

File: tests_integration/testing_utils/assertions/test_result_assertions.py (hunk 1 above)
Comment (sfc-gh-astus) [reply tests_integration/testing_utils/assertions/test_result_assertions.py]: Right, removed

**Review by sfc-gh-pjob — APPROVED** [review]

**Pull request merged**